bootstrap/handlers/httpserver.go

```diff
@@ -144,10 +144,14 @@ func (b *HttpServer) BootstrapHandler(
 		if err != nil && err != http.ErrServerClosed {
 			// Other errors occur during bootstrapping, like port bind fails, are considered fatal
 			lc.Errorf("Web server failed: %v", err)
+
+			// Allow any long-running go functions that may have started to stop before exiting
 			cancel := container.CancelFuncFrom(dic.Get)
-			cancel() // this will clean up any long-running go functions that may have started
-			// Give time for clean up to occur before exiting.
-			time.Sleep(1 * time.Millisecond)
+			cancel()
+
+			// Wait for all long-running go functions to stop before exiting.
+			wg.Done()
```

Comment:
This struck me as odd, as usually you don't see a call to `wg.Done` right before a `wg.Wait`. Expanding the code a bit, it looks like you are doing this because this go-routine's `defer` which calls `wg.Done` won't yet have been called, correct?

 Can you add a comment on why it is needed?

Reply:
Yep, the wait will block if we don't do this since this is for the current go func. 
Yep will add a comment.